udata/api/__init__.py

```diff
@@ -258,17 +258,57 @@ def handle_unauthorized_file_type(error):
     return {"message": msg}, 400
 
 
-validation_error_fields = api.model("ValidationError", {"errors": fields.Raw})
+field_validation_error_fields = api.model(
+    "FieldValidationError",
+    {"errors": fields.Raw, "message": fields.String},
+)
+validation_error_fields = api.model(
+    "ValidationError",
+    {"errors": fields.Raw, "message": fields.String},
+)
+
+validation_error_fields_v2 = apiv2.model(
+    "ValidationError",
+    {"errors": fields.Raw, "message": fields.String},
+)
 
 
-@api.errorhandler(FieldValidationError)
-@api.marshal_with(validation_error_fields, code=400)
-def handle_validation_error(error: FieldValidationError):
-    """A validation error"""
+def convert_object_of_exceptions_to_object_of_strings(exceptions: dict):
     errors = {}
-    errors[error.field] = [error.message]
+    for key, exception in exceptions.items():
+        if isinstance(exception, Exception):
+            errors[key] = str(exception)
+        elif isinstance(exception, dict):
+            errors[key] = convert_object_of_exceptions_to_object_of_strings(exception)
+        elif isinstance(exception, str):
+            errors[key] = exception
+        else:
+            log.warning(
+                f"Unknown type in `convert_object_of_exceptions_to_object_of_strings`: {exception}"
+            )
+            errors[key] = str(exception)
+
+    return errors
+
+
+@api.errorhandler(mongoengine.errors.ValidationError)
+@api.marshal_with(validation_error_fields, code=400)
+def handle_validation_error(error: mongoengine.errors.ValidationError):
+    """Error returned when validation failed."""
+    return (
+        {
+            "errors": convert_object_of_exceptions_to_object_of_strings(error.errors),
+            "message": str(error),
```

Comment:
Do we have only duplicated info between `errors` and `message`?

Reply:
Mostly yes. Message is formatted different for `FieldValidationError` and `ValidationError`